### PR TITLE
Initial commit of script to run cppcheck.

### PR DESCRIPTION
**Pull request by mmitche**
This script runs the cppcheck and sloccount tools for static analysis.  The lab runs this script with the default arguments.

/cc @janhenke

### CODE REVIEW

File: run-cppcheck.sh [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+    usage
+    exit 1
+fi
+
+check_dependencies
+
+if [ "$RunCppCheck" == true ]
+then
+    echo "Running cppcheck for files: $Files"
+    cppcheck --enable=all -j 5 --xml --xml-version=2 --force $Files 2> $CppCheckOutput
```
Comment (janhenke) [inline run-cppcheck.sh]: We should make a runtime detection of the number of cores here and do not hard code to 5.
From build.sh:
```
    # Get the number of processors available to the scheduler
    # Other techniques such as `nproc` only get the number of
    # processors available to a single process.
    if [ `uname` = "FreeBSD" ]; then
	NumProc=`sysctl hw.ncpu | awk '{ print $2+1 }'`
    else
	NumProc=$(($(getconf _NPROCESSORS_ONLN)+1))
    fi
```

**Comment by janhenke** [comment]
Aside from the comment, LGTM.

File: run-cppcheck.sh (hunk 1 above)
Comment (mmitche) [inline run-cppcheck.sh]: Missed that, thanks will change.

**Comment by mmitche** [comment]
@janhenke fixed, thanks for the review!

**Pull request merged**